Modules/Core/Transform/include/itkResampleInPlaceImageFilter.h

```diff
@@ -0,0 +1,160 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#ifndef itkResampleInPlaceImageFilter_h
+#define itkResampleInPlaceImageFilter_h
+
+#include "itkImageToImageFilter.h"
+#include "itkVersorRigid3DTransform.h"
+
+namespace itk
+{
+/** \class ResampleInPlaceImageFilter
+ * \brief Resample an image in place.
+ *
+ * The current ITK resample image filter will generate a physical memory-
+ * modified version of the input image if the input transform is not identity. The
+ * abuse use of the filter can be cumbersome in a situation that the image is very
+ * large, and there are lots of transform to be superimposed for the input image, and
+ * we even don't care about those intermediate transformed images.
+ *
+ * If all the transforms are rigid, a far superior way to achieve a similar result
+ * while minimizing the accumulative resampling errors as well as eliminating the expense
+ * on accessing the physical memory of the image is to compose all the
+ * transforms before hand if it is possible, and then we only need to resample the
+ * input image with the final composed transform once.
+ *
+ * Here we present a more compact alternative, all information is stored in the header
+ * of the image and there is no need to maintain the final transform any longer. ITK
+ * image class has innate support for doing this.
+ *
+ * \param RigidTransform -- Currently must be a VersorRigid3D
```

Comment:
Maybe `VersorRigid3D` can be prefixed with the appropriate Doxygen keyword to generate the link to the class documentation?

Reply:
I don't know off the bat what keyword is required. Maybe `\class`?

Reply:
I think `\class` would make it: https://www.doxygen.nl/manual/commands.html